test/src/unit-regression2.cpp

```diff
@@ -785,6 +791,7 @@ TEST_CASE("regression tests 2")
     }
 
 #if JSON_HAS_FILESYSTEM || JSON_HAS_EXPERIMENTAL_FILESYSTEM
+    // JSON_HAS_CPP_17 (do not remove; see note at top of file)
```

Comment:
I begin to wonder whether it would make more sense to teach CMake to know that `JSON_HAS_FILESYSTEM` implies `JSON_HAS_CPP_17` rather than to add magic words and justifying remarks.

Reply:
I thought about that as well. Either option seems at least a little bit messy. I'll see what it would take.

Reply:
At least it would be messy at one place rather than spread around multiple unit tests.

Reply:
Fair point. :-)

Reply:
Added to the list.